### ISSUE TITLE
New react-router version breaks with `TypeError: Cannot read property 'history' of undefined`

### ISSUE THREAD

**Issue by magneticflux-** [sanfilippopablo/react-swipeable-routes#17]
After updating to `react-router@4.4.0`, my app fails to run with this error:
```
Uncaught TypeError: Cannot read property 'history' of undefined
    at SwipeableRoutes.render (index.js:183)
    at finishClassComponent (react-dom.development.js:15276)
    at updateClassComponent (react-dom.development.js:15231)
    at beginWork (react-dom.development.js:16221)
    at performUnitOfWork (react-dom.development.js:20241)
    at workLoop (react-dom.development.js:20282)
    at HTMLUnknownElement.callCallback (react-dom.development.js:147)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:196)
    at invokeGuardedCallback (react-dom.development.js:250)
    at replayUnitOfWork (react-dom.development.js:19465)
    at renderRoot (react-dom.development.js:20395)
    at performWorkOnRoot (react-dom.development.js:21319)
    at performWork (react-dom.development.js:21229)
    at performSyncWork (react-dom.development.js:21203)
    at requestWork (react-dom.development.js:21058)
    at scheduleWork (react-dom.development.js:20871)
    at scheduleRootUpdate (react-dom.development.js:21566)
    at updateContainerAtExpirationTime (react-dom.development.js:21592)
    at updateContainer (react-dom.development.js:21660)
    at ReactRoot.push../node_modules/react-dom/cjs/react-dom.development.js.ReactRoot.render (react-dom.development.js:21973)
    at react-dom.development.js:22125
    at unbatchedUpdates (react-dom.development.js:21448)
    at legacyRenderSubtreeIntoContainer (react-dom.development.js:22121)
    at Object.render (react-dom.development.js:22196)
    at Module../src/index.tsx (index.tsx:14)
    at __webpack_require__ (bootstrap:781)
    at fn (bootstrap:149)
    at Object.0 (serviceWorker.ts:143)
    at __webpack_require__ (bootstrap:781)
    at checkDeferredModules (bootstrap:45)
    at Array.webpackJsonpCallback [as push] (bootstrap:32)
    at main.chunk.js:1
```
as well as this warning just before:
```
Warning: Failed context type: The context `router` is marked as required in `SwipeableRoutes`, but its value is `undefined`.
```

From `react-router`'s [api docs](https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/context.router.md):
> `context.router` should not be considered public API. Since context itself is an experimental API and may change in a future release of React, you should avoid accessing this.context.router directly in your components.

As a stop-gap measure until the usage of `context.router` can be replaced, I recommend adding the version qualifier `<4.4` to the peer dependency on `react-router` so that at least people are notified about the incompatibility in some way.

**Comment by HadrienPierart** [comment]
I'm experiencing the same issue.
Please dont enforce a max version of React Router, as v5 is expected to be backward compatible with v4.

**Comment by sanfilippopablo** [comment]
Done and published under `react-swipeable-routes@0.6.0-beta`. Can you try it? I'll do a proper release tomorrow

**Comment by HadrienPierart** [comment]
Hi there !
Thanks for the update, it works very nicely now :slightly_smiling_face: 
I started a fork on my own yesterday and ended up at the exact same code as you when it worked again and was about to push it when I saw your answer this morning :smile: 

Here is what I ended up with :  

```
const SwipeablePaths = [
  '/route1',
  '/route2',
  '/route3',
  '/route4',
];

const Swipeables = () => (
  <SwipeableRoutes>
    <Route exact path={'/route1'} component={Route1}/>
    <Route exact path={'/route2'} component={Route2}/>
    <Route exact path={'/route3'} component={Route3}/>
    <Route exact path={'/route4'} component={Route4}/>
  </SwipeableRoutes>
);

// In my render :
      <Router>
        <Suspense fallback={<div><img src={logo}/></div>}>
          <Switch>
            <Route exact path="/" component={() => <Redirect to={'/route1'} />}/>
            <Route exact path={SwipeablePaths} component={Swipeables}/>

            <Route exact path={"/route5"} component={Route5}/>
            <Route component={NotFoundRoute}/>
          </Switch>
        </Suspense>
      </Router>
```

I can confirm that this now works with a nice way of doing a 404, multiple routes in and out of swipeable, and also with React Suspense / Lazy ! Well done. (I also added a redirect of / onto one of the swipeable routes, because it was too easy otherwise :smile:  )

you just forgot a console.log in your render function

**Comment by sanfilippopablo** [comment]
That's great. I published 0.6.0 with the `console.log` removed. Thanks!

**Comment by magneticflux-** [comment]
Version `0.6.0` seems to work for me! Thanks!
